File: Linear Search/Linear Search.py
def linearsearch(arr,x):
    """

    Parameters
    ----------
    arr : List.
    x : Integer.
    
    Linear Search program to search any element in given list
    
    Returns
    -------
    Returns a Integer value of index of the element.

    """
    a =len(arr)
    for i in range(a):
        if arr[i] == x:
            return i
    return -1

def main():
    result = 0
    n = int(input("\nEnter the no. of Element in List : "))
    
    # Defining of list
    arr = list()
    for i in range(n):
        a = int(input("\nEnter the ELement : "))
        arr.append(a)
        
    #Element to find
    x = int(input("Enter the Element want to search in list : "))
    
    # Calling linear search
    result = linearsearch(arr,x)
    if result == -1:
        print("Element is not found in array")
    else:
        print("Element is found at index {}".format(result+1))

# Main Function Calling
if  __name__ == "__main__":
    main()

File: Linear Search/Linear Search.cpp
#include<iostream>
using namespace std;

// Linear Function
int main(int arr[], int x, int n)
{
    for (int i = 0; i <= n; i++)
    {
        // Searching for element
        if(arr[i] == x)
        {
            return i;
        }
    }
    return -1;
}

// Main Function
int main()
{
    // Declaring variables
    int n;
    int x;
    int result = 0;

    cout << "Enter the no. of element in array : "<<endl;
    cin >> n;
    
    int arr[n];
    while (i < n)
    {
        cout << "\nEnter the element : ";
        cin >> arr[i];
        i++;
    }
    
    cout << "\nEnter the Element want to search in array : ";
    cin >> x;
    
    // Calling Linear Search Function
    result = linearsearch(arr,x,n);
    if(result == -1)
    {
        cout << "\nElement is not found in array";
    }
    else
    {
        cout << "Element is found at index "<<result<<endl;
    }
}